File: M8/Exemplos/App_eXperience_Java/app/src/main/java/com/elgin/elginexperience/shipay/ShipayClient.java
package com.elgin.elginexperience.shipay;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ShipayClient {
    private ShipayService apiService;

    public ShipayService getApiService() {

        // Initialize ApiService if not initialized yet
        if (apiService == null) {
            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(Constants.BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

            apiService = retrofit.create(ShipayService.class);
        }

        return apiService;
    }
}

File: Exemplos/App_eXperience_Kotlin/E1_KOTLIN_R/app/src/main/java/com/example/e1_kotlin_r/shipay/ShipayClient.kt
package com.example.e1_elgin_kotlin.shipay

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


class ShipayClient {

    private lateinit var apiService: ShipayService

    fun getApiService() : ShipayService{

        if (!::apiService.isInitialized) {
            val retrofit = Retrofit.Builder()
                .baseUrl(Constants.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build()
            apiService = retrofit.create(ShipayService::class.java)
        }

        return apiService


    }


}

